numpy/ma/core.py

```diff
@@ -1248,7 +1248,7 @@ def _recursive_make_descr(datatype, newtype=bool_):
     # Is this some kind of composite a la (np.float,2)
     elif datatype.subdtype:
         mdescr = list(datatype.subdtype)
-        mdescr[0] = newtype
+        mdescr[0] = _recursive_make_descr(datatype.subdtype[0], newtype)
```

Comment:
@charris Because a [subdtype](http://docs.scipy.org/doc/numpy/reference/generated/numpy.dtype.subdtype.html) is a tuple `(item_dtype, shape)`.  The shape does not change, but the `item_dtype` does.